onnxruntime/test/contrib_ops/reverse_op_test.cc

```diff
@@ -0,0 +1,248 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "test/providers/provider_test_utils.h"
+#include "core/util/math.h"
+
+namespace onnxruntime {
+namespace test {
+
+template <typename T>
+void RunReverseTest(const std::initializer_list<T>& input_vals,
+                    const std::vector<int64_t>& input_dimensions,
+                    const std::vector<int64_t>& reverse_axes,
+                    const std::initializer_list<T>& expected_vals,
+                    const std::vector<int64_t>& expected_dimensions,
+                    OpTester::ExpectResult expect_result = OpTester::ExpectResult::kExpectSuccess,
+                    const std::string& expected_err_str = "") {
+  OpTester test("Reverse", 1, onnxruntime::kMSDomain);
+
+  // Attribute
+  if (reverse_axes.size() != 0)
+    test.AddAttribute("axes", reverse_axes);
+
+  // Input
+  test.AddInput<T>("input", input_dimensions, input_vals);
+
+  // Output
+  test.AddOutput<T>("output", expected_dimensions, expected_vals);
+
+  // Run test and check results
+  test.Run(expect_result, expected_err_str);
+}
+
+template <typename T>
+void RunReverseTestWrapper() {
+  // Test 1 (scalar)
+  std::initializer_list<T> input_vals_1 = {1};
+  std::vector<int64_t> input_dimensions_1;
+  std::vector<int64_t> reverse_axes_1;
+  std::initializer_list<T> expected_vals_1 = {1};
+  std::vector<int64_t> expected_dimensions_1;
+  RunReverseTest<T>(input_vals_1, input_dimensions_1, reverse_axes_1, expected_vals_1, expected_dimensions_1);
+
+  // Test 2 (default axes)
+  std::initializer_list<T> input_vals_2 = {1, 2, 3, 4, 5, 6, 7, 8};
+  std::vector<int64_t> input_dimensions_2 = {2, 4};
+  std::vector<int64_t> reverse_axes_2;
+  std::initializer_list<T> expected_vals_2 = {8, 7, 6, 5, 4, 3, 2, 1};
+  std::vector<int64_t> expected_dimensions_2 = {2, 4};
+  RunReverseTest<T>(input_vals_2, input_dimensions_2, reverse_axes_2, expected_vals_2, expected_dimensions_2);
+
+  // Test 3 (explicit axes)
+  std::initializer_list<T> input_vals_3 = {1, 2, 3, 4, 5, 6, 7, 8};
+  std::vector<int64_t> input_dimensions_3 = {2, 4};
+  std::vector<int64_t> reverse_axes_3 = {0, 1};
+  std::initializer_list<T> expected_vals_3 = {8, 7, 6, 5, 4, 3, 2, 1};
```

Comment:
Would be good to add tests that reverse a subset of axes to prove that works as well.

Reply:
Added such a test. Thanks.